certified-connectors/DocuSignDemo/script.csx

```diff
@@ -2118,13 +2118,221 @@ public class Script : ScriptBase
         templateRoles.Add(signer);
         signer = new JObject();
       }
+    }
+
+    newBody["templateRoles"] = templateRoles;
+
+    if (!string.IsNullOrEmpty(query.Get("status")))
+    {
+      newBody["status"] = query.Get("status");
+    }
 
-      if (key.Contains(" Signing Group"))
+    return newBody;
+  }
+
+  private JObject CreateEnvelopeFromTemplateV3BodyTransformation(JObject body)
```

Comment:
This whole method is not in a readable format. Please break it down to multiple methods. It will be hard to maintain this kind of code.

Reply:
Done

Reply:
Thank you